parsers/julia.c

```diff
@@ -0,0 +1,1400 @@
+            }
+        }
+    }
+    /* Pass the last backtick */
+    advanceAndStoreChar(lexer);
+}
+
+/* Parse a block with opening and closing character */
+static void scanBlock (lexerState *lexer, int open, int close, bool convert_newline)
+{
+    /* Assume the current char is `open` */
+    int level = 1;
+
+    /* Pass the first opening */
+    advanceAndStoreChar(lexer);
+
+    while (lexer->cur_c != EOF && level > 0)
+    {
+        /* Parse everything */
+        if (lexer->cur_c == ' ' || lexer->cur_c == '\t')
+        {
+            skipWhitespace(lexer, false);
+            vStringPut(lexer->token_str, ' ');
+        }
+        if (lexer->cur_c == '#')
+        {
+            skipComment(lexer);
+        }
+        else if (lexer->cur_c == '\"')
+        {
+            scanString(lexer);
+        }
+        else if (lexer->cur_c == '\'')
+        {
+            scanCharacterOrTranspose(lexer);
+        }
+
+        /* Parse opening/closing */
+        if (lexer->cur_c == open)
+        {
+            level++;
+        }
+        else if (lexer->cur_c == close)
+        {
+            level--;
+        }
+
+        if (convert_newline && skipNewLine(lexer))
+        {
+            vStringPut(lexer->token_str, ' ');
+        }
+        else
+        {
+            advanceAndStoreChar(lexer);
+        }
+
+    }
+    /* Lexer position is just after `close` */
+}
+
+
+/* Parse a block inside parenthesis, for example a function argument list */
+static void scanParenBlock (lexerState *lexer)
+{
+    scanBlock(lexer, '(', ')', true);
+}
+
+/* Indexing block with bracket.
+ * Some keywords have a special meaning in this environment:
+ * end, begin, for and if */
+static void scanIndexBlock (lexerState *lexer)
+{
+    scanBlock(lexer, '[', ']', false);
+
+}
+
+/* Parse a block inside curly brackets, for type parametrization */
+static void scanCurlyBlock (lexerState *lexer)
+{
+    scanBlock(lexer, '{', '}', true);
+}
+
+/* Scan type annotation like
+ * `::Type`, `::Type{T}`
+ */
+static void scanTypeAnnotation (lexerState *lexer)
+{
+    /* assume that current char is '<', '>' or ':', followed by ':' */
+    advanceAndStoreChar(lexer);
+    advanceAndStoreChar(lexer);
+
+    skipWhitespace(lexer, true);
+    scanIdentifier(lexer, false);
+    if (lexer->cur_c == '{')
+    {
+        scanCurlyBlock(lexer);
+    }
+}
+
+/* Scan type annotation like
+ * `where Int<:T<:Real`, `where S<:Array{Real}` or `where {S, T}`
+ */
+static void scanTypeWhere (lexerState *lexer)
+{
+    /* assume that current token is 'where'
+     * allow line continuation */
+    vStringPut(lexer->token_str, ' ');
+    skipWhitespace(lexer, true);
+
+    while (lexer->cur_c != EOF)
+    {
+
+        if (lexer->cur_c == '{')
+        {
+            scanCurlyBlock(lexer);
+        }
+        else if (isIdentifierFirstCharacter(lexer->cur_c))
+        {
+            scanIdentifier(lexer, false);
+            if (endswith(lexer->token_str->buffer, "where"))
```

Comment:
buffer is a private field of vString.
Could you use vStringValue(lexer->token_str) instead of acessing buffer field directly.